### ISSUE TITLE
Support `drop default index`

### ISSUE THREAD

**Issue by frankmcsherry** [MaterializeInc/materialize#15079]
### Feature request

It is easy to type `create default index on <thing>`. However, if you realize you don't want that, it takes more knowledge to figure out how to undo what you have done. You have to learn about `show indexes on <thing>` and potentially sort out which one was the one you added (e.g. if there was already another index). Supporting `drop default index` could make this less complicated.

**Comment by benesch** [comment]
The wrinkle with this is that _we_ also lose track of whether you typed `CREATE DEFAULT INDEX` or not. `CREATE DEFAULT INDEX` is pure syntax sugar over:

```
CREATE INDEX $RELATION_primary_idx ON $RELATION (col1, col2, ..., colN)
```

Say you run the following sequence of commands:

```
CREATE TABLE t (a int, b int);
CREATE INDEX t_primary_idx ON t (a);
CREATE INDEX t_primary_idx1 ON t (b);
-- The next command is equivalent to: CREATE INDEX t_primary_idx2 ON t (a, b);
CREATE DEFAULT INDEX ON t; 
CREATE INDEX t_primary_idx3 ON t (a, b);
-- The next command is equivalent to: CREATE INDEX t_primary_idx4 ON t (a, b);
CREATE DEFAULT INDEX ON t; 
DROP DEFAULT INDEX t;
```

What does `DROP DEFAULT INDEX t` do? Does it drop `t_primary_idx`? `t_primary_idx2? `t_primary_idx4`? Both `t_primary_idx2` and `t_primary_idx4`? 

To make this work sensibly, we might need to record whether an index was created by `CREATE DEFAULT INDEX` or not, and only let you have one index created by `CREATE DEFAULT INDEX` at a time.

**Comment by morsapaes** [comment]
I agree with Nikhil's point on syntax sugar. In the end (and if memory doesn't fail me), when `CREATE MATERIALIZED VIEW` was sugar for `CREATE VIEW`+`CREATE DEFAULT INDEX`, we also dropped these using `DROP VIEW` and only introduced dedicated syntax once we turned materialized views into an actual relation type.

> However, if you realize you don't want that, it takes more knowledge to figure out how to undo what you have done.

Wouldn't documenting the naming pattern used for default indexes possibly address your concern, @frankmcsherry? I can't seem to find the trace back to where we abandoned the [idea of dropping](https://materializeinc.slack.com/archives/C015RHB3LDR/p1658931799402109) `CREATE DEFAULT INDEX` altogether, but if this isn't even something we expect users to type frequently, it'd also be a good reason to not introduce new syntax!
